cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -483,6 +483,30 @@ std::string FormatRange(int64_t start, int64_t length) {
   return ss.str();
 }
 
+// An AWS RetryStrategy that wraps a provided arrow::fs::S3RetryStrategy
+class WrappedRetryStrategy : public Aws::Client::RetryStrategy {
+ public:
+  explicit WrappedRetryStrategy(const std::shared_ptr<S3RetryStrategy>& s3_retry_strategy)
+      : s3_retry_strategy_(s3_retry_strategy) {}
+
+  bool ShouldRetry(const Aws::Client::AWSError<Aws::Client::CoreErrors>& error,
+                   long attempted_retries) const override {  // NOLINT
```

Comment:
sorry, what's the NOLINT for? Nothing stands out to me as violating standards

Reply:
Most of this class was copy-pasted from [ConnectRetryStrategy](https://github.com/apache/arrow/blob/0d5e4550f3fc39797590f79fa03a9895bd7163c8/cpp/src/arrow/filesystem/s3_internal.h#L193), which included these NOLINTs. I'll see if I can get rid of them